components/eventing-controller/controllers/subscription/nats/reconciler_test.go

```diff
@@ -71,76 +70,93 @@ var (
-			subscriptionName := fmt.Sprintf(subscriptionNameFormat, id) + "-valid"
-			subscriberName := fmt.Sprintf(subscriberNameFormat, id) + "-valid"
-
+		It("should mark the Subscription as not ready until NATS is reachable again", func() {
 			ctx := context.Background()
 			natsPort := natsPort + id
 			natsServer, natsURL := startNATS(natsPort)
 			defer reconcilertesting.ShutDownNATSServer(natsServer)
 			cancel = startReconciler(eventTypePrefix, natsURL)
 			defer cancel()
 
-			// create subscriber svc
+			// Create a subscriber service
+			subscriberName := fmt.Sprintf(subscriberNameFormat, id) + "-valid"
 			subscriberSvc := reconcilertesting.NewSubscriberSvc(subscriberName, namespaceName)
 			ensureSubscriberSvcCreated(ctx, subscriberSvc)
 
-			// create subscription
-			subscription := reconcilertesting.NewSubscription(subscriptionName, namespaceName,
+			// Create a subscription
+			subscriptionName := fmt.Sprintf(subscriptionNameFormat, id) + "-valid"
+			subscription := reconcilertesting.NewSubscription(subscriptionName, subscriberSvc.Namespace,
 				reconcilertesting.WithFilter("", eventTypeToSubscribe),
-				reconcilertesting.WithValidSink(namespaceName, subscriberName),
+				reconcilertesting.WithSinkURLFromSvc(subscriberSvc),
 			)
 			ensureSubscriptionCreated(ctx, subscription)
 
-			getSubscription(ctx, subscription).Should(And(
-				reconcilertesting.HaveSubscriptionName(subscriptionName),
-				reconcilertesting.HaveCondition(eventingv1alpha1.MakeCondition(
+			Context("testing against an available NATS server", func() {
+				expectedConditions := eventingv1alpha1.MakeCondition(
 					eventingv1alpha1.ConditionSubscriptionActive,
 					eventingv1alpha1.ConditionReasonNATSSubscriptionActive,
-					v1.ConditionTrue, "")),
-			))
+					v1.ConditionTrue, "")
+				getSubscription(ctx, subscription).Should(And(
+					reconcilertesting.HaveSubscriptionName(subscriptionName),
+					reconcilertesting.HaveCondition(expectedConditions),
+				))
+			})
 
-			natsServer.Shutdown()
-			getSubscription(ctx, subscription, timeout, pollingInterval).Should(And(
-				reconcilertesting.HaveSubscriptionName(subscriptionName),
-				reconcilertesting.HaveSubscriptionNotReady()),
-			)
+			Context("testing against an unavailable NATS server", func() {
+				natsServer.Shutdown()
+				getSubscription(ctx, subscription, timeout, pollingInterval).Should(And(
+					reconcilertesting.HaveSubscriptionName(subscriptionName),
+					reconcilertesting.HaveSubscriptionNotReady()),
+				)
+			})
 
-			_, _ = startNATS(natsPort)
-			getSubscription(ctx, subscription, timeout, pollingInterval).Should(And(
-				reconcilertesting.HaveSubscriptionName(subscriptionName),
-				reconcilertesting.HaveSubscriptionReady()),
-			)
+			Context("testing against a NATS server that is available again", func() {
+				_, _ = startNATS(natsPort)
+				getSubscription(ctx, subscription, timeout, pollingInterval).Should(And(
```

Comment:
We should. Unfortunately, because some of our test functions like `startReconciler` only work if called from inside of an `It`, with the current setup it works best if the whole test is in one big `It`. I think this should be one of our goals of these refactorings. Maybe think of it as an To-Do for refactoring iteration 3 or so.